packages/common/src/ArcGISContext.ts

```diff
@@ -280,6 +302,47 @@ export class ArcGISContext implements IArcGISContext {
     if (opts.properties) {
       this._properties = opts.properties;
     }
+
+    if (opts.environment) {
+      this._environment = opts.environment;
+    } else {
+      let environment: HubEnvironment = "production";
+      const portalSubdomain =
+        this._portalSelf?.portalHostname.split(".")[0] ||
+        getPortalBaseFromOrgUrl(this._portalUrl);
+      const portalSubdomainHash = {
+        devext: "devext",
+        qaext: "qaext",
+        www: "production",
+        // enterprise: "enterprise",
+        // sandbox: "sandbox",
+        // localdev: "localdev"
+      };
+      environment = (portalSubdomainHash as any)[portalSubdomain];
+      if (this._portalSelf.isPortal) {
+        environment = "enterprise";
+      }
+      this._environment = environment;
+    }
```

Comment:
Rough implementation: if we are passed environment on options, use it, if not, infer it from the info we have.